index.js

```diff
@@ -51,17 +41,46 @@ class Doc extends EventEmitter {
       this[kGC] = new GCMetric()
     }
 
-    // TODO: maybe here is better to use a Map
-    this[kData] = {
-      raw: {}
-    }
     this[kSample]()
-    const timer = setInterval(this[kEmitStats].bind(this), this[kOptions].sampleInterval)
-    timer.unref()
+    if (this[kOptions].autoStart) {
+      this.start()
+    }
+  }
+
+  start () {
+    this[kTimer] = setInterval(this[kEmitSample].bind(this), this[kOptions].sampleInterval)
+    if (this[kOptions].unref) {
+      this[kTimer].unref()
+    }
+  }
+
+  stop () {
+    clearTimeout(this[kTimer])
+  }
+
+  get cpu () {
+    return this[kCpu]
   }
 
-  [kEmitStats] () {
-    this.emit('data', this[kSample]())
+  get eventLoopDelay () {
+    return this[kEventLoopDelay]
+  }
+
+  get gc () {
+    return this[kGC]
+  }
+
+  get activeHandles () {
+    return this[kActiveHandles]
+  }
+
+  get memory () {
+    return this[kMemory]
+  }
+
+  [kEmitSample] () {
+    this[kSample]()
+    this.emit('sample')
```

Comment:
I noticed that we're no longer emitting a `data` object here in the event.

Is the intended way to consume this to listen for the event, and then just read from the instance itself?

```js
doc.on('sample', () => {
  console.log(doc.cpu.usage)
})
```

?

Reply:
Yes, it is. I did this to expose the metrics as both computed and raw (where possible) and let the user choose what to use. This is the best way I could come up with without having a huge object as a payload in the event handler.

Reply:
Another reason behind this change is the stateful nature of the event loop Histogram instance. I was afraid it could have caused unwanted behavior. Forcing the user to extract the data at a certain point in time is a more explicit and coherent strategy IMHO.

Reply:
Yep, that makes sense to me. As long as it's clear in the README I'm fine with using it that way. :)